velox/functions/remote/if/CMakeLists.txt

```diff
@@ -14,7 +14,11 @@
 
 include(FBThriftCppLibrary)
 
-add_fbthrift_cpp_library(velox_remote_function_thrift RemoteFunction.thrift)
+add_fbthrift_cpp_library(velox_remote_function_thrift RemoteFunction.thrift
+                         SERVICES RemoteFunctionService)
+
+target_compile_options(velox_remote_function_thrift
+                       PRIVATE -Wno-deprecated-declarations)
```

Comment:
nit: `-Wno-deprecated-declarations` can go to the main CMakeLists.txt under ENABLE_ALL_WARNINGS

Reply:
Sorry to disagree but in modern CMake warnings should be disabled for specific targets and we should ideally migrate away from modifying `CXX_FLAGS` directly. 

The target based approach prevents legitimate warnings from being suppressed through global flags by accident. `target_compile_options` also allows for the use of generator expressions to tailor the flags to specific compilers and versions.

Reply:
The nice part about adding this is here is that it's only disabled for this target, without potentially masking other legit warnings/errors in other targets throughout the library.

Reply:
Makes sense. We should fix modifying CXX_FLAGS soon so that handling warnings is uniform.

Reply:
I started doing that, but finding the right targets to place these flags is super tedious :(